nixos/modules/installer/tools/tools.nix

```diff
@@ -224,12 +224,23 @@ in
         # accidentally delete configuration.nix.
         # system.copySystemConfiguration = true;
 
-        # This value determines the NixOS release from which the default
-        # settings for stateful data, like file locations and database versions
-        # on your system were taken. It's perfectly fine and recommended to leave
-        # this value at the release version of the first install of this system.
-        # Before changing this value read the documentation for this option
-        # (e.g. man configuration.nix or on https://nixos.org/nixos/options.html).
+        # This option defines the first version of NixOS you have installed on this particular machine,
+        # and is used to maintain compatibility with data not managed by NixOS configuration (e.g. databases)
+        # that was created by applications running on older NixOS versions.
+        #
+        # Most users should NEVER change this value after the initial install, for any reason,
+        # even if you've upgraded your system.
+        #
+        # This value does NOT affect the Nixpkgs version your packages and OS are pulled from,
+        # so changing it will NOT upgrade your system.
+        #
+        # This value being lower than the current NixOS release does NOT mean your system is
+        # out of date, out of support, or vulnerable.
+        #
+        # Do NOT change this value unless you have manually inspected all the changes it would make to your configuration,
+        # and migrated your data accordingly.
+        #
+        # For more information, see `man configuration.nix`.
```

Comment:
I think this is good and clear. A few thoughts:
* "This option defines the first version of NixOS you have installed on this particular machine" kinda blurs the difference between the current install and previous installs of nixos on the same machine. It’s hopefully not a confusion anyone runs into, but it seems like an unneccessary confusion. Maybe "This option is neccessary to remember the version of NixOS used during initial install on this particular machine."?

* "with data not managed by NixOS configuration" this is confusing. The data is not generated from NixOS configuration, but file location, services, etc. are managed by NixOS configuration. We are looking for a beginner friendly term for "stateful data", right? How about simply "compatibility with application data (e.g. databases) that was created by older NixOS versions."

* For my personal taste, this comment is too long. I would confine more of the warnings into the option documentation. That’s exactly why the comment previously said "a) just leave this, it’s fine and b) before changing read the docs".  In the end I have no clue what works better for users. Just saying, that we maybe need a different solution before half of configuration.nix is an essay about stateVersion.^^

* Why drop the the nixos.org link? I assume for many users it is the more natural interface to read options than the man-page which is arguably only semi-made for this.

I would love to see my first two points considered somehow, the second two are nice to have.

Reply:
1. I get the problem, but I think "initial install" can be misread mostly the same way. I can't really think of a good way to phrase this to make it clear we mean "the first version of this _install_" without using "install" is a noun, which is confusing.

2. Yes, that was the idea. I think I like your version, though "application data" is technically never created by NixOS, it's created by applications - but that's nitpicky and not relevant to the target audience for this comment.

3. Originally it was intro, 3 points, see also. Now it's 4 points (with #3 being added on suggestion from @tejing), and I think this is also my upper limit for how many points it should have. I'd like to condense it more, but I think more points that are punchier individually is better than less points that don't hit as hard.

4. No particular reason, I think it just got lost in the sauce. I'll put it back.

Reply:
If this were a book I would be very opposed to the one paragraph - one sentence policy. I think I would probably just pack the points 2 to 5 in one paragraph. That way we have three paragraphs "What is this, what to do, further reading". But that’s really just my intuition. You can leave it as is.

Reply:
I actually intentionally separated these into individual lines, because I want them to scan as individual points. For a book, you'd probably want to write the whole thing very differently. This is basically our equivalent to the ["this is not a place of honor" messages](https://en.wikipedia.org/wiki/Long-term_nuclear_waste_warning_messages).

Reply:
Applied [2] and [4]. Still thinking of a good way to phrase [1].